File: LeetCode/Medium/0204-count-primes/0204-count-primes.py
import math

class Solution(object):
    def create(self, n):
        self.box = [True] * n
        self.box[0],self.box[1] = False,False

        for i in range(2,int(n**0.5+1)):
            if self.box[i]:
                for j in range(i*i,n,i):
                    self.box[j] = False

    def countPrimes(self, n):
        """
        :type n: int
        :rtype: int
        """
        if(n<=1): 
            return 0

        self.create(n)

        return sum(self.box)
       

File: LeetCode/Medium/0204-count-primes/0204-count-primes.cpp
class Solution {
public:
    int countPrimes(int n) {
        if(n<=1) return 0;

        vector<int>box(n,1);

        box[0] = 0; box[1] = 0;

        for(int i = 2; i*i<n; i++)
            if(box[i])
                for(int j=i*i; j < n; j+=i)
                    box[j] = 0;
       
        int cnt = 0;
        for(int i=0; i<n; i++)
            cnt += box[i];

        return cnt;
    }
};